### PR TITLE
build updates, package upgrades

### PR DESCRIPTION
**Pull request by keymone**
- drop trusty
- bump pyyaml, kazoo, paasta-tools, service config lib and cryptography
- switch to python 3.7
- use latest dh_virtualenv to simplify dockerfiles

### CODE REVIEW

**Review by vkhromov — APPROVED** [review]

File: dockerfiles/xenial/Dockerfile [hunk 1]
```diff
@@ -22,31 +20,19 @@ RUN apt-get update && apt-get -y install \
     libyaml-dev \
     libzookeeper-mt-dev \
     protobuf-compiler \
-    # dh-virtualenv only works with python2 - https://github.com/spotify/dh-virtualenv/issues/122
-    python \
-    python3.6-dev \
+    python3.7-dev \
+    python3.9-dev \
```
Comment (Rob-Johnson) [inline dockerfiles/xenial/Dockerfile:24]: why do we need 3.9 too?

**Review by Rob-Johnson — APPROVED** [review]

File: dockerfiles/xenial/Dockerfile (hunk 1 above)
Comment (keymone) [reply dockerfiles/xenial/Dockerfile:24]: dependency of latest dh_virtualenv which is a deb build for ubuntu 22

Comment (keymone) [reply dockerfiles/xenial/Dockerfile]: dropped it as it works with 3.7 too and i'm force-installing the deb anyway

**Comment by keymone** [comment]
itests pass and package_* targets finish succesfully, i've installed new packages on couple hosts for basic checks and everything seems to work, going to merge and cut a new release

**Comment by keymone** [comment]
shipped in e5ce9b9

**Pull request closed without merging**